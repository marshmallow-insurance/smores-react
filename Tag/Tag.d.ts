import { FC } from 'react';
import { MarginProps } from '../utils/space';
import { Color } from '../theme';
export type TagProps = {
    label: string;
    color: Color;
    bgColor: Color;
    borderColor?: Color;
    className?: string;
} & MarginProps;
export declare const Tag: FC<TagProps>;
