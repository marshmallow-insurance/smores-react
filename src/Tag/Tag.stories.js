import React from 'react'
import { storiesOf } from '@storybook/react'
import { text } from '@storybook/addon-knobs'

import { Tag } from './Tag'
import { theme } from '../theme'

storiesOf('Tag', module).addWithJSX('default', () => (
  <Tag
    label={text('Label', 'Covered')}
    bgColor={theme.colors.green[500]}
    borderColor={theme.colors.green[700]}
    color={text('Color', 'white')}
    className={text('Class name', '')}
  />
))
