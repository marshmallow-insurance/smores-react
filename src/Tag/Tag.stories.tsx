import React from 'react'
import { Tag, TagProps } from './Tag'
import { CollectionPage } from './Collection'

export default {
  title: 'Tag',
  component: Tag,
}

const Template = (props: TagProps) => <Tag {...props} />

export const Default = Template.bind({})

Default.args = {
  label: 'This is a tag',
  bgColor: 'feijoa',
  borderColor: 'feijoa',
  color: 'cream',
}

Default.argTypes = {
  theme: { table: { disable: true } },
  as: { table: { disable: true } },
  forwardedAs: { table: { disable: true } },
}

export const Collection = CollectionPage.bind({})
