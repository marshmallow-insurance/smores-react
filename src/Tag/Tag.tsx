import React, { FC } from 'react'
import styled, { useTheme } from 'styled-components'
import { MarginProps } from '../utils/space'

import { TransientProps } from 'utils/utilTypes'
import { Box } from '../Box'
import { Text } from '../Text'
import { Icon } from '../Icon'
import { Icons } from 'Icon/iconsList'
import {
  ColorTypes,
  getColorPath,
  resolveToThemeColor,
} from '../ThemeProvider/utils/colourMap'

export type TagProps = {
  label: string
  color: ColorTypes
  bgColor: ColorTypes
  borderColor?: ColorTypes
  className?: string
  icon?: Icons
  iconColor?: ColorTypes
  onClick?: () => void
} & MarginProps

export const Tag: FC<TagProps> = ({
  label,
  color,
  borderColor,
  bgColor,
  className,
  icon,
  iconColor,
  onClick,
  ...marginProps
}) => {
  const theme = useTheme()

  const resolvedBGColor = resolveToThemeColor(bgColor, theme)
  const resolvedBorderColor =
    borderColor && resolveToThemeColor(borderColor, theme)

  return (
    <Wrapper
      $bgColor={resolvedBGColor}
      className={className}
      $borderColor={resolvedBorderColor}
      {...marginProps}
      alignContent="center"
      justifyContent="center"
      onClick={onClick}
    >
      {icon && (
        <TagIcon
          render={icon}
          color={iconColor}
          size={16}
          data-testid={`tag-icon-${icon}`}
        />
      )}
      <TagText tag="span" typo="label" color={getColorPath(color)}>
        {label}
      </TagText>
    </Wrapper>
  )
}

type WrapperProps = TransientProps<{
  bgColor: string
  borderColor?: string
}>

const Wrapper = styled(Box)<WrapperProps>`
  background-color: ${({ $bgColor }) => $bgColor};
  border: ${({ $borderColor }) => $borderColor && `1px solid ${$borderColor}`};

  border-radius: 6px;
  padding: 4px 8px;

  height: 22px;

  box-sizing: border-box;
  display: inline-flex;
`

const TagText = styled(Text)`
  display: flex;
  align-items: center;
  font-size: 10px;
  font-weight: 500;
  line-height: 12px;
  text-transform: uppercase;
`

const TagIcon = styled(Icon)`
  padding-right: 4px;
`
