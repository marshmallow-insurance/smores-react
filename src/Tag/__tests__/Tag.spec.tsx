import { render } from '@testing-library/react'
import React from 'react'

import { Tag } from '../Tag'

test('renders', () => {
  const { container } = render(
    <Tag
      label="Cancelled - Refunded Manually"
      color="strawberry"
      bgColor="cream"
    />,
  )
  expect(container.firstChild).toMatchSnapshot()
})
