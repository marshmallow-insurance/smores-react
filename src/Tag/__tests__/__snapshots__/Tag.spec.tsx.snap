// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders 1`] = `
<div
  class="sc-beySPh jlZyMd sc-eDLKkx bCShqs"
>
  <span
    class="sc-beySPh kknHUq sc-hLQSwg cibKFK sc-jTQCzO bEAEJu"
    cursor="inherit"
    title=""
  >
    Cancelled - Refunded Manually
  </span>
</div>
`;
